sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -0,0 +1,506 @@
+        "disbursementValue.valueType",
+        "disbursementValue.valueCode",
+        "disbursementValue.valueAmount",
+        "disbursementValue.disbursedAmountSubtracted",
+        "studentAssessment.id",
+      ])
+      .addSelect(
+        `CASE
+            WHEN EXISTS(${this.studentRestrictionService
+              .getExistsBlockRestrictionQuery(
+                false,
+                false,
+                "restrictionActionType",
+              )
+              .getSql()}) THEN true
+            ELSE false
+        END`,
+        "stopFullTimeBCFunding",
+      )
+      .innerJoin("disbursement.studentAssessment", "studentAssessment")
+      .innerJoin("studentAssessment.application", "application")
+      .innerJoin("application.currentAssessment", "currentAssessment") // * This is to fetch the current assessment of the application, even though we have multiple reassessments
+      .innerJoin("currentAssessment.offering", "offering")
+      .innerJoin("offering.institutionLocation", "institutionLocation")
+      .innerJoin("offering.educationProgram", "educationProgram")
+      .innerJoin("application.student", "student") // ! The student alias here is also used in sub query 'getExistsBlockRestrictionQuery'.
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .innerJoin("application.msfaaNumber", "msfaaNumber")
+      .innerJoin("disbursement.disbursementValues", "disbursementValue")
+      .where("disbursement.dateSent is null")
+      .andWhere("disbursement.disbursementDate <= :disbursementMinDate", {
+        disbursementMinDate,
+      })
+      .andWhere("application.applicationStatus = :applicationStatus", {
+        applicationStatus: ApplicationStatus.completed,
+      })
+      .andWhere("msfaaNumber.dateSigned IS NOT NULL")
+      .andWhere("msfaaNumber.cancelledDate IS NULL")
+      .andWhere("sinValidation.isValidSIN = true")
+      .andWhere("offering.offeringIntensity = :offeringIntensity", {
+        offeringIntensity,
+      })
+      .andWhere(
+        `NOT EXISTS(${this.studentRestrictionService
+          .getExistsBlockRestrictionQuery()
+          .getSql()})`,
+      )
+      .andWhere(
+        "disbursement.disbursementScheduleStatus = :disbursementScheduleStatus",
+        {
+          disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+        },
+      )
+      .andWhere("disbursement.coeStatus = :coeStatus", {
+        coeStatus: COEStatus.completed,
+      })
+      .setParameter("restrictionActionType", stopFullTimeBCFunding)
+      // 'restrictionActions' parameter used inside sub-query.
+      .setParameter("restrictionActions", possibleRestrictionActions)
+      // Necessary, for instance, for the overawards processing.
+      .orderBy("disbursement.disbursementDate")
+      .getRawAndEntities();
+    return mapFromRawAndEntities<ECertDisbursementSchedule>(
+      queryResult,
+      "stopFullTimeBCFunding",
```

Comment:
Is it required/significant to return this when the offering intensity is part time?

Reply:
Nope, but I was not willing to do the change now and I end up restricting the method `shouldStopFunding`.
This query was kept almost as it was, it was just moved from one service to another and added a few conditions.

Reply:
May be a todo?

Reply:
As we talked about, this query is likely to be changed, and changing it right now is not in the scope of the ticket.
Either way, I will leave a todo.